go/obscuronode/enclave/sql/edgelessdb.go

```diff
@@ -0,0 +1,523 @@
+     - prepare a manifest.json that contains that CA cert and some SQL to initialise the DB tables and user
+     - submit that manifest.json file to {edbAddress}/manifest, using the certificate provided from /quote to authenticate
+     - seal and persist the manifest.json and the certs so we can reconnect if enclave is restarted
+
+   When connecting to an already-initialized edgeless DB we must:
+     - perform remote attestation on the edgeless db
+     - unseal the manifest.json and get the hash of it, also unseal the certificate that edb was initialized with
+	 - verify the /signature request for edgeless DB matches the manifest.json hash
+     - connect to edb with the persisted cert - it's now safe to read and write to the DB
+
+	Some useful documentation for this:
+		Main edb docs: https://docs.edgeless.systems/edgelessdb/#/
+		EDB demo docs: https://github.com/edgelesssys/edgelessdb/tree/main/demo
+		// Note: due to an issue with the dependency, I've duplicated the relevant parts of the ERA tool into edb_attestation.go
+		ERA - remote attestation tool: https://github.com/edgelesssys/era
+*/
+
+const (
+	edbHTTPPort          = "8080"
+	edbManifestEndpoint  = "/manifest"
+	edbSignatureEndpoint = "/signature"
+
+	dataDir         = "/data"
+	certIssuer      = "obscuroCA"
+	certSubject     = "obscuroUser"
+	enclaveHostName = "enclave"
+
+	dbUser    = "obscuro"
+	dbName    = "obsdb"
+	tableName = "keyvalue"
+	keyCol    = "ky"
+	valueCol  = "val"
+
+	// The attestation config comes from here (https://github.com/edgelesssys/edgelessdb/releases/latest/download/edgelessdb-sgx.json)
+	//     todo: revisit whether we want this hardcoded
+	edbAttestationConf = "{\n\t\"SecurityVersion\": 2,\n\t\"ProductID\": 16,\n\t\"SignerID\": \"67d7b00741440d29922a15a9ead427b6faf1d610238ae9826da345cea4fee0fe\"\n}"
+
+	// change this flag to true to debug issues with edgeless DB (and start EDB process with -e EDG_EDB_DEBUG=1
+	//   this will give you:
+	//  	- verbose logging on EDB
+	//		- write the edb.pem file out for connecting to Edgeless DB services manually
+	//		- versions of files created with a '.unsealed' suffix that can be used to connect to the database using mysql-client
+	debugMode = false
+)
+
+var (
+	edbCredentialsFilepath = filepath.Join(dataDir, "edb-credentials.json")
+	attestationCfgFilepath = filepath.Join(dataDir, "edgelessdb-sgx.json")
+
+	manifestSQLStatements = []string{
+		fmt.Sprintf("CREATE USER %s REQUIRE ISSUER '/CN=%s' SUBJECT '/CN=%s'", dbUser, certIssuer, certSubject),
+		fmt.Sprintf("CREATE DATABASE %s", dbName),
+		fmt.Sprintf("CREATE TABLE %s.%s (%s varbinary(64) primary key, %s blob)", dbName, tableName, keyCol, valueCol),
+		fmt.Sprintf("GRANT ALL ON %s.%s TO %s", dbName, tableName, dbUser),
+	}
+)
+
+type manifest struct {
+	SQL   []string `json:"sql"`
+	Cert  string   `json:"ca"`
+	Debug bool     `json:"debug"`
+}
+
+type EdgelessDBConfig struct {
+	Host string
```

Comment:
let's add a todo here to add more settings here like the attestationConf, usernames, etc